store/store.go

```diff
@@ -1174,6 +1174,25 @@ func (s *Store) Query(qr *proto.QueryRequest) ([]*proto.QueryRows, error) {
 	if qr.Level == proto.QueryRequest_QUERY_REQUEST_LEVEL_NONE && s.isStaleRead(qr.Freshness, qr.FreshnessStrict) {
 		return nil, ErrStaleRead
 	}
+
+	if qr.LeaderReadOpt {
```

Comment:
Now that I think about it, this should be:
`if proto.QueryRequest_QUERY_REQUEST_LEVEL_STRONG && qr.LeaderReadOpt()`

In other words the flag should only have an effect if strong is selected.